### ISSUE TITLE
"no warnings or errors encountered even though `delayed_good_path_bugs` issued" after panic

### ISSUE THREAD

**Issue by RalfJung** [rust-lang/rust#117381]
I'm getting a panic
```
thread 'rustc' panicked at compiler/rustc_ty_utils/src/abi.rs:360:21:
`PassMode::Direct` for aggregates only allowed on wasm and `extern "ptx-kernel"` fns
Problematic type: TyAndLayout {
    ty: ReprC1<*const i32>,
...
```
(that's expected, I'm working on that part of the compiler)
and after that I am getting a second error
```
error: internal compiler error: no warnings or errors encountered even though `delayed_good_path_bugs` issued

error: internal compiler error: trimmed_def_paths constructed but no error emitted; use `DelayDm` for lints or `with_no_trimmed_paths` for debugging
```
That's bad, it makes the panic message harder to find. It's also wrong, there was an error encountered, a panic even!

Cc @compiler-errors  -- that's one of the trimmed-path issues I encountered recently (but might be unrelated to https://github.com/rust-lang/rust/pull/117373)
